### ISSUE TITLE
Doing gradient descent in function spaces

### ISSUE THREAD

**Issue by Seanny123** [ctn-waterloo/modelling_ideas#66]
In a [Request for Research from OpenAI](https://openai.com/requests-for-research/#natural-q-learning), they suggest applying Natural Gradient Descent to Q-Learning. They describe Natural Gradient Descent as follow:

> We can identify a neural network with its parameters, and use the backpropagation algorithm to slowly change the parameters to minimize the cost function. But we can also think of a neural network as of a high-dimensional manifold in the infinite-dimensional space of all possible functions, and we can, at least conceptually, run gradient descent in function space, subject to the constraint that we stay on the neural network manifold.

According to the Request for Research, this is actually related to Hessian-Free-Learning and a bunch of other stuff I don't understand because I still haven't learned how to math good. However, the terms "High Dimensional", "Function Space" and "Q-Learning" seemed to hint to me that this might be a possible solution to the problem of "the NEF can't do Backprop". That being said, I've only spent 10 minutes investigating this, so I'm only 10% sure this is actually applicable, but wanted to share it anyways.
